homeassistant/components/ollama_conversation/models.py

```diff
@@ -0,0 +1,46 @@
+"""Models for Ollama conversation integration."""
+
+from dataclasses import dataclass
+from enum import StrEnum
+
+import ollama
+
+from homeassistant.core import State
+
+
+class MessageRole(StrEnum):
+    """Role of a chat message."""
+
+    SYSTEM = "system"  # prompt
+    USER = "user"
+
+
+@dataclass
+class MessageHistory:
+    """Chat message history."""
+
+    timestamp: float
+    """Timestamp of last use in seconds."""
+
+    messages: list[ollama.Message]
+    """List of message history, including system prompt and assistant responses."""
+
+    @property
+    def num_user_messages(self) -> int:
+        """Return a count of user messages."""
+        return sum(1 if m["role"] == MessageRole.USER else 0 for m in self.messages)
```

Comment:
Booleans are numbers
```suggestion
        return sum(m["role"] == MessageRole.USER for m in self.messages)
```

Reply:
Haskell would like to have a word :smile: